File: front-pagos/src/app/models/Usuario.ts
export interface Usuario {
  dni: string;
  nombreCompleto: string;
  password?: string;
  rol?: string;
}

export interface RegisterUsuarioRequest {
  dni: string;
  nombreCompleto: string;
  password: string;
  rol: string;
}

export interface EditUsuarioRequest {
  nombreCompleto?: string;
  password?: string;
  rol?: string; 
}

export interface LoginRequest {
  dni: string;
  password: string;
}

export interface LoginResponse {
  dni: string;
  nombreCompleto: string;
  rol?: string;
}

File: Pagos-Backend/src/main/java/com/registro/pagos/models/Usuario.java
package com.registro.pagos.models;

import jakarta.persistence.*;

@Entity
@Table(name = "usuarios")
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String dni;

    @Column(nullable = true)
    private String passwordHash;

    @Column(nullable = true)
    private String nombreCompleto;

    @Column
    private String rol;

    public Usuario() {}

    public Usuario(String dni, String passwordHash, String nombreCompleto, String rol) {
        this.dni = dni;
        this.passwordHash = passwordHash;
        this.nombreCompleto = nombreCompleto;
        this.rol = rol;
    }

    // getters y setters
    public Long getId() { return id; }
    public String getPasswordHash() { return passwordHash; }
    public void setPasswordHash(String passwordHash) { this.passwordHash = passwordHash; }
    public String getNombreCompleto() { return nombreCompleto; }
    public void setNombreCompleto(String nombreCompleto) { this.nombreCompleto = nombreCompleto; }
    public void setId(Long id) {
        this.id = id;
    }

    public String getRol() {
        return rol;
    }

    public void setRol(String rol) {
        this.rol = rol;
    }

    public String getDni() {
        return dni;
    }

    public void setDni(String dni) {
        this.dni = dni;
    }
}